### ISSUE TITLE
xml to json conversion using pyang & xsltproc

### ISSUE THREAD

**Issue by abhijitbhalekar** [mbj4668/pyang#193]
Converted xml netconf reply to json using pyang. Following is the json output which seems wrong as per yang json format. x-module prefix should not be included for child "upstream-data-rate-profile" as it is already included for its parent "configs".

{
"x-module:configs": {
"profiles": {
"x-module:profile": [
{ "name": "default-profile", "maximum": 50000, "maximum-rate": 50000, "minimum": 2000, "minimum-rate": 2000 }
]
}
}
}

**Comment by llhotka** [comment]
Which version of pyang do you use? I can't reproduce the problem, it seems to work OK:
```
$ cat test.yang
module test {
  namespace "http://example.com/test";
  prefix t;

  container configs {
    container profiles {
      list profile {
        key "name";
        leaf name {
          type string;
        }
      }
    }
  }
}
$ cat test-data.xml 
<?xml version='1.0' encoding='UTF-8'?>
<data xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <configs xmlns="http://example.com/test">
    <profiles>
      <profile>
	<name>default-profile</name>
      </profile>
    </profiles>
  </configs>
</data>
$ pyang -f jsonxsl -o test.xsl test.yang 
$ xsltproc test.xsl test-data.xml
{
  "test:configs": {
    "profiles": {
      "profile": [
        {
          "name": "default-profile"
        }
      ]
    }
  }
}
```

**Comment by abhijitbhalekar** [comment]
I am using pyang 1.5.
Also, in my example, profiles is augmented and is defined in a submodule. Something like as follows,

module x-base {
  namespace "http://example.com/x-base";
  prefix x-b; 
  /* Common */
  include x-base2;
   /* Profiles */
  include x-profiles;
}

submodule x-base2 {
  belongs-to x-base {
    prefix x-b;
  }
  container configs {
    container profiles {
    }
  }
}

submodule x-profiles {
  belongs-to x-base {
    prefix x-b;
  }

  augment "/x-b:configs/x-b:profiles" {      
    list profile {
      key name;
      leaf name {
            type String;
      }
    }
  }
}

**Comment by llhotka** [comment]
Pyang 1.6 considers these modules invalid. Okay, `String` should be `string`, but pyang also cannot find the `x-b:configs` node which looks like a bug. I will open a new issue for it, and come back to this after it is resolved.

BTW, if you include code snippets like you modules, wrap them in triple backquotes so that they will be rendered as my example above.

**Comment by llhotka** [comment]
Sorry, for some reason I lost the `include x-base2;` statement in `x-profiles.yang`. With it the modules are valid. Issue #194 still applies to YANG 1.1 though.

**Comment by abhijitbhalekar** [comment]
Any update on this issue? Are you able to reproduce it?

**Comment by llhotka** [comment]
As you can see above, it was fixed in commit 124d56d. So please test it with your data, and then we can close the issue.

**Comment by abhijitbhalekar** [comment]
Issue is resolved. Now json output is without x-module prefix for the child if parent has already included it. Closing the issue. Thanks for the fix.
